src/pages/TxSender.vue

```diff
@@ -49,69 +49,134 @@
 	</a-layout-content>
 </template>
 
-<script>
+<script lang="ts">
 import { defineComponent, createVNode } from "vue";
 import { ExclamationCircleOutlined } from "@ant-design/icons-vue";
 import { Modal } from "ant-design-vue";
 import WalletStore from "@/store/WalletStore";
-import { isJson } from "../utilities/validate";
+import {
+	isJson,
+	convertBase64ToUnit8Array,
+	formatJSON,
+	convertObjectValuesToUnit8Array,
+	convertToBase64,
+} from "@/utilities";
+import algosdk from "algosdk";
+import {
+	errorMessage,
+	loadingMessage,
+	openErrorNotificationWithIcon,
+	openSuccessNotificationWithIcon,
+	successMessage,
+	TRANSACTION_SEND_SUCCESSFUL,
+} from "@/constants";
+import { WalletType } from "@/types";
 
 export default defineComponent({
 	data() {
 		return {
 			isJsonSelected: true,
-			txInput: "",
+			txInput: undefined,
 			txOutput: "",
 			txInputError: "",
+			key: "SenderKey",
 		};
 	},
 	setup() {
 		const walletStore = WalletStore();
-		const sendTx = async () => {
-			console.log("send transaction");
+		return {
+			walletStore,
 		};
-		const confirm = () => {
+	},
+	methods: {
+		confirm() {
 			Modal.confirm({
 				title: "Confirm",
 				icon: createVNode(ExclamationCircleOutlined),
 				okText: "Ok",
 				cancelText: "Cancel",
 				centered: true,
-				onOk: sendTx,
+				onOk: this.sendTx,
 			});
-		};
-		return {
-			confirm,
-		};
-	},
-	methods: {
+		},
+		async sendTx() {
+			loadingMessage(this.key);
+			let encodedTx: Uint8Array | string;
+			if (!this.txInput) return;
+			// algosigner accepts base64 not unit8Array
+			if (this.walletStore.walletKind === WalletType.ALGOSIGNER) {
+				if (this.isJsonSelected) {
+					encodedTx = convertToBase64(
+						convertObjectValuesToUnit8Array(JSON.parse(this.txInput)["blob"])
+					);
+				} else encodedTx = this.txInput;
+			} else {
+				if (this.isJsonSelected) {
+					// decode blob back to unit8Array
+					encodedTx = convertObjectValuesToUnit8Array(
+						JSON.parse(this.txInput)["blob"]
+					);
+				} else encodedTx = convertBase64ToUnit8Array(this.txInput);
+			}
+			this.walletStore.webMode
+				.sendAndWait(encodedTx)
+				.then((response: any) => {
```

Comment:
response type is TxConfirmedInfo I think

Reply:
yes it is will update